src/osiris_log.erl

```diff
@@ -1884,6 +1821,61 @@ open_index_read(File) ->
     _ = file:read(Fd, ?IDX_HEADER_SIZE),
     Fd.
 
+scan_idx(IndexFile, Offset, LastChunkInSegment) when is_list(IndexFile) ->
```

Comment:
I'm thinking it would be neat if this function just took the `seg_info` record which contains the index file and the last chunk info.

Reply:
then it would differ in arity to to the main "worker" function which is a bit clearer.

Reply:
also we could guard against offset requests that are out of bounds